### ISSUE TITLE
Pointers on Train_Step_And_Loss.ipynb

### ISSUE THREAD

**Issue by sayakpaul** [ayulockin/SwAV-TF#1]
- We need to trace what is non-trainable and what is. In the main SwAV code, all the variables with `no_grad` tag are basically non-trainable. 
- Did not get this part - "## crossentropy loss between code and p, assuming that code is to be precited from the assigned cluster. if wrong then logits will be label and vice versa". Are you meaning that the `logits` and `labels` in criterion will be swapped? If so, how?
- I think it might be even better to just replicate the following as the authors have done in [here](https://github.com/facebookresearch/swav/blob/master/main_swav.py#L310-L314). What do you think?

  ![image](https://user-images.githubusercontent.com/22957388/89723899-053b5200-da1a-11ea-8ad1-b5742a90bb5d.png)

- Two normalizations. First, they normalize the embeddings they get from the RN50 backbone, then they pass it through a linear layer (prototype). While training they again normalize this prototype vector.
